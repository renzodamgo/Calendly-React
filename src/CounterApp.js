import React, {useState} from 'react'
import PropTypes from 'prop-types'
import { InlineWidget, PopupWidget, PopupButton } from "react-calendly";


const CounterApp = ({value = 10})=>{
  const [counter, setCounter] = useState(value);

  // Sumar value
  const handleAdd = (e)=>{
    // console.log(e);
    // setCounter(counter + 1);
    setCounter((c)=> c + 1)
  }

  const handleSubstract = (e)=>{
    setCounter((c)=> c - 1)
  }

  const handleReset = (e) => {
    setCounter((c)=> c = value)
  }


  return (
    <>
      <h1>Demo Calendly y Active Campaign</h1>
      <InlineWidget url="https://calendly.com/rebels247" />
      <PopupWidget url="https://calendly.com/rebels247" />
      <PopupButton url="https://calendly.com/rebels247"/>
    </>

  )
}

CounterApp.propTypes={
  value: PropTypes.number,
}

export default CounterApp