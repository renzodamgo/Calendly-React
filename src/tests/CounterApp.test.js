import { shallow } from "enzyme";
import CounterApp from "../CounterApp";

describe('Pruebas en CounterApp', () => {
    let wrapper = shallow(<CounterApp />);

    // Antes de cada prueba se reinicia el componente
    beforeEach(()=>{
        wrapper = shallow(<CounterApp />);
    })

    test('Debería renderizar Counter App Correctamente', () => {
        expect(wrapper).toMatchSnapshot()

    });

    test('Debería renderizar en 100 el contador', () => {
        const value = 100;
        const wrapper = shallow(<CounterApp value={value} />)
        const NumeroContador = wrapper.find('h2').text().trim();
        expect(NumeroContador).toBe('100')
    });

    test('Debe de incrementar con el botton +1', () => {
        wrapper.find('button').at(0).simulate('click');
        const NumeroContador = wrapper.find('h2').text().trim();
        expect(NumeroContador).toBe('11')
    });

    // EL valor es 11 porque corrío la prueba anterior

    test('Debe de decrecer con el botton -1', () => {
        //Con esto se reinicia el componente a su estado original
        // const wrapper = shallow(<CounterApp />)

        wrapper.find('button').at(1).simulate('click');
        // console.log(btn.html())
        const NumeroContador = wrapper.find('h2').text().trim();
        expect(NumeroContador).toBe('9')
    });

    test('Debe de resetear con el botton reset', () => {
        const wrapper = shallow(<CounterApp value={105}/>)
        wrapper.find('button').at(1).simulate('click');
        wrapper.find('button').at(1).simulate('click');
        wrapper.find('button').at(2).simulate('click');
        const NumeroContador = wrapper.find('h2').text().trim();
        expect(NumeroContador).toBe('105')
    });
})
