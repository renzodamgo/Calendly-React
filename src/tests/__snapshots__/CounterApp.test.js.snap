// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp Deber√≠a renderizar Counter App Correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
     +1 
  </button>
  <button
    onClick={[Function]}
  >
     -1 
  </button>
  <button
    onClick={[Function]}
  >
     Reset 
  </button>
</Fragment>
`;
